base/tks/src/main/java/org/dogtagpki/server/tks/rest/v2/TPSConnectorServlet.java

```diff
@@ -0,0 +1,175 @@
+//
+// Copyright Red Hat, Inc.
+//
+// SPDX-License-Identifier: GPL-2.0-or-later
+//
+package org.dogtagpki.server.tks.rest.v2;
+
+import java.io.PrintWriter;
+import java.net.URLEncoder;
+import java.util.stream.Collectors;
+
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.http.HttpSession;
+
+import org.dogtagpki.server.tks.rest.base.TPSConnectorProcessor;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.netscape.certsrv.base.WebAction;
+import com.netscape.certsrv.key.KeyData;
+import com.netscape.certsrv.system.TPSConnectorCollection;
+import com.netscape.certsrv.system.TPSConnectorData;
+import com.netscape.certsrv.util.JSONSerializer;
+
+/**
+ * @author Marco Fargetta {@literal <mfargett@redhat.com>}
+ */
+@WebServlet(
+        name = "tksTPSConnector",
+        urlPatterns = "/v2/admin/tps-connectors/*")
+public class TPSConnectorServlet extends TKSServlet {
+    private static final long serialVersionUID = 1L;
+    private static final Logger logger = LoggerFactory.getLogger(TPSConnectorServlet.class);
+
+    private TPSConnectorProcessor tpsConnector;
+
+    @Override
+    public void init() throws ServletException {
+        super.init();
+        tpsConnector = new TPSConnectorProcessor(engine);
+    }
+
+    @WebAction(method = HttpMethod.GET, paths = {""})
+    public void findConnectors(HttpServletRequest request, HttpServletResponse response) throws Exception {
+        HttpSession session = request.getSession();
+        logger.debug("TPSConnectorServlet.findConnectors(): session: {}", session.getId());
+        int size = request.getParameter("pageSize") == null ?
+                DEFAULT_SIZE : Integer.parseInt(request.getParameter("pageSize"));
+        int start = request.getParameter("start") == null ? 0 : Integer.parseInt(request.getParameter("start"));
+        String host = request.getParameter("host");
+        String port = request.getParameter("port");
+        TPSConnectorCollection connectors = tpsConnector.findConnectors(host, port, start, size);
+        PrintWriter out = response.getWriter();
+        out.println(connectors.toJSON());
+    }
+
+    @WebAction(method = HttpMethod.POST, paths = {""})
+    public void createConnector(HttpServletRequest request, HttpServletResponse response) throws Exception {
+        HttpSession session = request.getSession();
+        logger.debug("TPSConnectorServlet.createConnector(): session: {}", session.getId());
+        String host = request.getParameter("host");
+        String port = request.getParameter("port");
```

Comment:
Just a note, we probably should consider requiring the user to specify the connector ID so that we don't have to generate it.  The current `findNextConnectorID()` might not generate consistent IDs across the cluster which could be problematic.

Reply:
Yes, the current implementation of `findNextConnectorID()` does not work across a cluster. The user could provide the ID or we have to implement a way to get the information consistent in a cluster scenario. Do we create a ticket for this?

Reply:
This is a design issue, not just the API, but also the storage, which might not be trivial to fix. So far there's nobody asking to fix it so I'd just leave it as is for now. Maybe in the future when we fully move to containers this issue will come up again and we can redesign at that point.